### PR TITLE
Add Schema for OpenTSDB verification plugin

### PR DESCRIPTION
**Pull request by shashankrnr32**
# Summary

@yahoo/ychaos-dev

Introduce Schema for OpenTSDB Verification Plugin

---

**Fixes**: #{issue_number}

#### Prerequisites
- [x] I have read the contribution guidelines

#### Code Analysis
- [x] Covered by Unittests
- [ ] Security warnings ignored (Why?!)

#### Project related
- [x] Schema changed and is backward compatible.
- [ ] Introduces a new sub-command for ychaos CLI

### Autogenerated Files
- [x] Auto generated schema

### Pre-Merge Checklist

- [ ] All the build jobs are passing

---

### Checklist (Reviewer 1)

- [ ] Reviewed Source Code
- [ ] Reviewed Tests (If applicable)
- [ ] Reviewed Documentation (If applicable)

### Checklist (Reviewer 2)

- [x] Reviewed Source Code
- [x] Reviewed Tests (If applicable)
- [x] Reviewed Documentation (If applicable)

### CODE REVIEW

File: tests/resources/testplans/valid/testplan8.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                start: '30m-ago'
+                end: 'now'
+                queries:
+                    -   aggregator: zimsum
+                        metric: system.cpu.used
+                        downsample: 15m-avg-nan
+                        filters:
+                            -   type: literal_or
+                                tagk: host
+                                filter: mockhost.resilience.cloud
```
Comment (r-r-2) [inline tests/resources/testplans/valid/testplan8.yaml:22]: nit :
```suggestion
                                filter: mockhost.resilience.yahoo.cloud
```

**Review by r-r-2 — APPROVED** [review]
lgtm

File: src/ychaos/testplan/verification/__init__.py [hunk 2]
```diff
@@ -145,6 +144,56 @@ class NoOpConfig(SchemaModel):
...
+        description=(
+            "Metrics verification criteria without state information."
+            "All the criteria part of this list must pass for the verification to be successful"
+        ),
+    )
+
+    state_bound_criteria: List[StateBoundMetricsVerificationCriteria] = Field(
+        default=list(),
+        description=(
+            "Metrics verification criteria with state. "
```
Comment (AlfinST) [inline src/ychaos/testplan/verification/__init__.py:178]: nit:
keeping it consistent with the rest
```suggestion
            "Metrics verification criteria with state."
```

File: src/ychaos/testplan/verification/__init__.py [hunk 3]
```diff
@@ -155,7 +204,11 @@ class VerificationType(AEnum):
 
     PYTHON_MODULE = "python_module", SimpleNamespace(schema=PythonModuleVerification)
     HTTP_REQUEST = "http_request", SimpleNamespace(schema=HTTPRequestVerification)
-    SDV4_VERIFICATION = "sdv4", SimpleNamespace(schema=SDv4Verification)
+    SDV4_VERIFICATION = (
```
Comment (AlfinST) [inline src/ychaos/testplan/verification/__init__.py:207]: nit:
isn't it easier to read this if this is a single line?

File: src/ychaos/testplan/verification/plugins/metrics.py [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+
+class MetricsAggregator(AEnum):
+    """
+    The Metrics aggregator options. Allows the user to transform a
+    time series data into comparable data.
+    """
+
+    AVG = "avg", SimpleNamespace(
+        aggregate=TimeSeriesDataAggregator.avg,
+        __desc__="Gets the Average of all the datapoints",
```
Comment (AlfinST) [inline src/ychaos/testplan/verification/plugins/metrics.py:138]: nit:
```suggestion
        __desc__="Gets the average of all the datapoints",
```

File: src/ychaos/testplan/verification/__init__.py (hunk 3 above)
Comment (shashankrnr32) [reply src/ychaos/testplan/verification/__init__.py:207]: The formatting is based on the `black` rules. The code follows a strict word count per line. That is why

Comment (AlfinST) [reply src/ychaos/testplan/verification/__init__.py:207]: Pointed that out cause that line was the smallest in that block 😅

**Review by AlfinST — APPROVED** [review]
lgtm

File: src/ychaos/testplan/verification/__init__.py (hunk 3 above)
Comment (shashankrnr32) [reply src/ychaos/testplan/verification/__init__.py:207]: Yes. Agree. But i dont manually change the formatting of code. `black .` does it

**Review by r-r-2 — APPROVED** [review]

**Comment by shashankrnr32** [comment]
Merging this.

**Pull request merged**